Add motd docs

Shadowsniper784 opened the pull request:
add docs for unconnected ping and pong as the motd

Shadowsniper784 committed:
Create MOTD.md

Shadowsniper784 committed:
Fix formatting in MOTD

theaddonn commented:
THAT WAS FAST!🔥

Shadowsniper784 committed:
Fix missing OfflineMessageId

Shadowsniper784 committed:
Redo, add code examples

theaddonn commented on docs/MOTD.md:
```diff
@@ -0,0 +1,362 @@
+#### Field Breakdown & Data Types
+
+- **Edition:** string  
+  Specifies the game edition (e.g., `MCPE` for Minecraft Pocket Edition), seems to be hardcoded to `MCPE`
```
I'm not actually sure about that one, it might be that Education Edition uses `MCEE` instead of `MCPE`

theaddonn replied:
wiki.vg seems to say the same... 
![image](https://github.com/user-attachments/assets/612a8919-d522-4458-ba78-db297cf0a125)

theaddonn commented on docs/MOTD.md:
```diff
@@ -0,0 +1,362 @@
+- **Protocol Version:** uint16  
+  Numeric identifier indicating the game protocol version (e.g., `748`).  
+- **Game Version:** string  
+  Denotes the server's game version (e.g., `1.20.40`, or on CubeCraft: `1`).  
```
not directly an issue, but love how you mentioned it XD
I found the same thing a few months ago, it is quite hilarious... wonder if there is a good reason for it

theaddonn commented on docs/MOTD.md:
```diff
@@ -0,0 +1,362 @@
+
+## Code Examples
+
+::: code-group
```
wonder if there is a way to make this collapsible, since not everyone might want to see it directly..

theaddonn commented on docs/MOTD.md:
```diff
@@ -0,0 +1,362 @@
+# MOTD Protocol Documentation
```
maybe add what motd directly stands for.. (Message of the day)

theaddonn commented on docs/MOTD.md:
```diff
@@ -0,0 +1,362 @@
+
+When a client wants to display the server it sends an unconnected ping
+
+- **Packet ID (1 byte):** `0x01`
```
I wonder in general if we should use a schema like rust uses (`u32`) or more over the older C style (`unsigned int`) or something in between (`uint32`). I personally prefer something like `u32`...

theaddonn reviewed (changes requested):
Seems very good! But there are still a few things that might need an improvement..

oh, also move this into the [`docs/raknet/`](https://github.com/bedrock-crustaceans/protocol-wiki/tree/main/docs/raknet) directory, thanks :)

MisledWater79 commented on docs/MOTD.md:
```diff
@@ -0,0 +1,362 @@
+
+## Overview
+
+The server operates by listening on port `19132` for incoming RakNet unconnected pings. Upon receiving a valid ping, the server responds with an unconnected pong that contains the MOTD and other info. This interaction ensures that clients receive essential details about the server before establishing a connection.
```
Maybe rephrase to be:
```suggestion
A RakNet server, by default, listens on port `19132` for incoming [Unconnected Pings](#unconnected-ping). Once a valid ping is sent, the server sends back a [Unconnected Pong](#unconnected-pong). This contains the server's (MOTD)[#motd-format), along with other server info. All this ensures that a client receives the details of the server before a connection is established.
```

Shadowsniper784 committed:
Fix the intro of MOTD.md

Co-authored-by: MisledWater79 <59891258+MisledWater79@users.noreply.github.com>

Shadowsniper784 committed:
Fixed some issues

Moved to /raknet
Changed to Message Of The Day
Added Education Edition
Did not fix the datatyoes

theaddonn commented:
How are the changes going?

MisledWater79 commented:
Actually scratch that, thats a LOT of code... maybe move that to a link to github or different page.

MisledWater79 commented on docs/raknet/MOTD.md:
````diff
@@ -0,0 +1,364 @@
+listen(19132);
+```
+:::
+## Sources
+- [Go Raknet](https://github.com/Sandertv/go-raknet/blob/master/internal/message/unconnected_ping.go#L8)<br>
+- [PHP MC Query](https://github.com/xPaw/PHP-Minecraft-Query/blob/master/src/MinecraftQuery.php#L212)<br>
+- [Raknet](https://github.com/facebookarchive/RakNet/blob/1a169895a900c9fc4841c556e16514182b75faf8/Source/RakPeer.cpp#L135)<br>
+- [Phantom](https://github.com/jhead/phantom/blob/44056d83afbfe60b253faff01308d171ac21e8d6/internal/proto/proto.go#L23)<br>
+- [Geyser](https://github.com/GeyserMC/Protocol/blob/46b4ad37b159b1fc59e45871f7572101f5ed43ab/bedrock-connection/src/main/java/org/cloudburstmc/protocol/bedrock/BedrockPong.java#L23)<br>
+- [Wiki.vg](https://wiki.vg/Raknet_Protocol#Unconnected_Ping)<br>
+- Wireshark 
````
Use `#### Resources` instead, along with that set the links to be this format: `[[wiki.vg/Minecraft Wiki](https://minecraft.wiki/w/Minecraft_Wiki:Projects/wiki.vg_merge/Raknet_Protocol)]: Some description here of what was used` (I think the wiki.vg link is the only one that needs updated since it was moved to the actual wiki)

MisledWater79 commented on docs/raknet/MOTD.md:
```diff
@@ -0,0 +1,364 @@
+  Ends with `0` to signify the conclusion of the MOTD string.  
+
+## Code Examples
+::: details Code Examples
```
Remove the dropdown as ill be implementing it directly to the code groups.

MisledWater79 commented on docs/raknet/MOTD.md:
````diff
@@ -0,0 +1,364 @@
+  ```
+- **GUID (8 bytes, Big-Endian):** Unique identifier for the client.
+
+#### Validation Criteria
+
+1. **Packet Size:**  
+   The total size must be exactly 33 bytes.
+2. **Packet ID:**  
+   Must strictly be `0x01`.
+3. **Timestamp:**  
+   It is just a timestamp of uptime, but we can check it is not infront of unix timestamp
+4. **Magic Offline Message ID:**  
+   Must match the predefined 16-byte sequence.
+5. **GUID:**  
+   It doesn't really need to be checked, but it is there
````
I don't see a point to this area, personally I think it's not really useful. let me know if I'm wrong though.

MisledWater79 commented on docs/raknet/MOTD.md:
The explanaition of the packets shouldn't be in this file, as this is labeled `Message of the Day (MOTD) Protocol Documentation` so it should be more just on the motd and not explaining the entirety of where it comes from.

Shadowsniper784 replied:
Should I split it into separate files like: unconnectedpong.md or something like that? Maybe in like a MOTD subgroup as they aren't used elsewhere

MisledWater79 replied:
I would say keep all motd stuff in this file and either move it to and make the unconnected pong file, or just remove it and someone else could make it and reference this file

MisledWater79 commented on docs/raknet/MOTD.md:
````diff
@@ -0,0 +1,364 @@
+::: details Code Examples
+::: code-group
+
+```rust [src/main.rs]
````
I'd say change the names to just be a Rust and Typescript instead of a file path.
